test/integration/edge_connectivity/edge_connectivity_test.go

```diff
@@ -0,0 +1,421 @@
+// +build integration
+
+package edge_connectivity
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/kube"
+        cli "github.com/skupperproject/skupper/client"
+	"gotest.tools/assert"
+)
+
+/*================================================================
+  Multiple tests of different topologies, in each case with
+  a single edge.
+  Build the topology, then ask the edge about its connectivity.
+  See if it has the right number of direct and indirect
+  connections.
+================================================================*/
+
+var fp = fmt.Fprintf
+
+var verbose bool = true
+var red string = "\033[1;31m"
+var green string = "\033[1;32m"
+var cyan string = "\033[1;36m"
+var resetColor string = "\033[0m"
+
+type SkupperNamespace struct {
+	name   string
+	isEdge bool
+
+	client         *cli.VanClient
+	secretFileName string
+}
+
+type Connexion struct {
+	from string
+	to   string
+}
+
+type TopologyTest struct {
+	name             string
+	diagram          []string
+	namespaces       []*SkupperNamespace
+	connexions       []Connexion
+	direct, indirect int // These are the expected results.
+}
+
+func getNamespace(targetNamespace string, t *TopologyTest) *SkupperNamespace {
+	for _, ns := range t.namespaces {
+		if ns.name == targetNamespace {
+			return ns
+		}
+	}
+
+	fp(os.Stdout, "getNamespace error: can't find client |%s|\n", targetNamespace)
+	os.Exit(1)
+
+	return nil
+}
+
+func getEdge(t *TopologyTest) *SkupperNamespace {
+	for _, ns := range t.namespaces {
+		if strings.Contains(ns.name, "edge") {
+			return ns
+		}
+	}
+
+	return nil
+}
+
+func getConnectivity(
+	testName string,
+	namespace *SkupperNamespace,
+	ctx context.Context,
+	attempts int,
+	expectedDirect, expectedIndirect int) (direct, indirect int, err error) {
+
+	for {
+		if attempts <= 0 {
+			break
+		}
+		info(testName, "Checking connectivity...")
+		response, err := namespace.client.VanRouterInspect(ctx)
+
+		if err == nil {
+			// Sometimes it takes a while for all the connected sites
+			// to be reported. If we don't have them yet, keep waiting
+			// until we get them, or until we time out.
+			if response.Status.ConnectedSites.Direct >= expectedDirect &&
+				response.Status.ConnectedSites.Indirect >= expectedIndirect {
+				info(testName, "success.")
+				return response.Status.ConnectedSites.Direct,
+					response.Status.ConnectedSites.Indirect,
+					nil
+			}
+		}
+		if response != nil {
+			info(testName, fmt.Sprintf("%d : (%d,%d) != (%d,%d)", attempts, response.Status.ConnectedSites.Direct, response.Status.ConnectedSites.Indirect, expectedDirect, expectedIndirect))
+		}
+		time.Sleep(time.Second)
+		attempts--
+	}
+	return 0, 0, fmt.Errorf("timed out")
+}
+
+func check(t *testing.T, err error, test, msg string) {
```

Comment:
You can put t.Helper() here,  that, in case of error, marks the error in the line where this function was called, and not in this line which is not usefull at all.
```

// Helper marks the calling function as a test helper function.                      
// When printing file and line information, that function will be skipped.           
// Helper may be called simultaneously from multiple goroutines.
```

Reply:
xlnt.